Sources/Sentry/SentrySpan.m

```diff
@@ -84,10 +96,50 @@ - (instancetype)initWithContext:(SentrySpanContext *)context
         _spanId = context.spanId;
         _sampled = context.sampled;
         _origin = context.origin;
+
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+        if (SentrySDK.options.enableContinuousProfiling) {
+            _profileSessionIDs = [NSMutableSet<NSString *> set];
+            if (SentryContinuousProfiler.isCurrentlyProfiling) {
+                [_profileSessionIDs
+                    addObject:SentryContinuousProfiler.currentProfilerID.sentryIdString];
+            }
+            [SentryDependencyContainer.sharedInstance.notificationCenterWrapper
+                addObserver:self
+                   selector:@selector(linkProfiler)
+                       name:kSentryNotificationContinuousProfileStarted];
+            [SentryDependencyContainer.sharedInstance.notificationCenterWrapper
```

Comment:
`m`: Wouldn't subscribing to started be enough? If there is an ongoing profile running, this method already adds the profileID. Also, adding it when a profile stops would add it twice, or is there an edge case that I don't see?

Reply:
Yes, so I was thinking about the case where the profiler is already running by the time the span starts and then use the stop notification to get the profile ID, but if the span stops before the profile, and then it would never get that one either. We should just check for any current profilers running when the span starts and grab the profile ID at that time.